docs/tutorials/mastering.md

```diff
@@ -0,0 +1,618 @@
+- `access-ip`: the IP address for the UPF to use on the `access` subnet
+- `core-gateway-ip`: this is the IP address of the gateway that knows how to route traffic from the UPF towards the internet
+- `core-interface`: the name of the MACVLAN interface on the Kubernetes host cluster to bridge to the `core` subnet
+- `core-ip`: the IP address for the UPF to use on the `core` subnet
+- `gnb-subnet`: the subnet CIDR where the gNB radios are reachable.
+
+```{note}
+The `access-gateway-ip` is expected to forward the packets from the `access-interface` to the `gnb-subnet`
+```
+
+```console
+cat << EOF > upf-overlay.yaml
+applications:
+  upf:
+    options:
+      access-gateway-ip: 10.202.0.1
+      access-interface: access
+      access-ip: 10.202.0.10/24
+      core-gateway-ip: 10.203.0.1
+      core-interface: core
+      core-ip: 10.203.0.10/24
+      gnb-subnet: 10.204.0.0/24
+      external-upf-hostname: upf.mgmt
+EOF
+```
+
+Next, we create a Juju model to represent the User Plane, using the cloud `user-plane-cluster`, which was created in the Bootstrap step.
+
+```console
+juju add-model user-plane user-plane-cluster
+```
+
+Deploy user plane bundle:
+
+```console
+juju deploy sdcore-user-plane --trust --channel=edge --overlay upf-overlay.yaml
+```
+
+Now expose the UPF service offering with Juju.
+
+```console
+juju offer upf:fiveg_n4
+```
+
+### Checkpoint
+
+You should be able to see the UPF External load balancer service in Kubernetes. Log into the user plane cluster and execute the following command:
+
+```console
+microk8s.kubectl get services -A | grep LoadBalancer
+```
+
+This should produce output similar to the following indicating that the PFCP agent of the UPF is exposed on 10.201.0.200 UDP port 8805
+```
+user-plane  upf-external  LoadBalancer  10.152.183.126  10.201.0.200  8805:31101/UDP
+```
+
+## 4. Deploy the gNB Simulator
+
+Create a new Juju model named `gnbsim` on the Juju controller cloud:
+
+```console
+juju add-model gnbsim gnb-cluster
+```
+
+Deploy the simulator to the gnbsim cluster. The simulator needs to know the following:
+
+- `gnb-interface`: the name of the MACVLAN interface to use on the host
+- `gnb-ip-address`: the IP address to use on the gnb interface
+- `icmp-packet-destination`: the target IP address to ping. If there is no egress to the internet on your core network, any IP that is reachable from the UPF should work.
+- `upf-gateway`: the IP address of the gateway between the RAN and Access networks
+- `upf-ip-address`: the IP address of the UPF to use
+- `imsi`: the International Mobile Subscriber Identity (IMSI) of the UE that is being simulated
+
+```console
+juju deploy sdcore-gnbsim gnbsim --trust --channel=edge \
+--config gnb-interface=ran \
+--config gnb-ip-address=10.204.0.10/24 \
+--config icmp-packet-destination=8.8.8.8 \
+--config upf-gateway=10.204.0.1 \
+--config upf-subnet=10.202.0.0/16 \
+--config imsi=208930100007487
```

Comment:
Do we need to configure IMSI (especially that this is the default value anyway)?
And if we do, then why don't we also set other params (like MCC, MNC, etc.)?

Reply:
We don't.  I was going to add a section where we specify an IMSI that is not provisioned and walk through the authentication failure use case, but not sure if we should as part of the learning.